dashboard/app/models/sections/section.rb

```diff
@@ -578,7 +568,36 @@ def update_code_review_expiration(enable_code_review)
   end
   before_validation :strip_emoji_from_name
 
-  public def add_instructor(email, current_user)
+  # Adds an instructor to the section
+  # If the instructor was previously deleted, restore the instructor
+  # If the instructor was invited or inactive, reactive the instructor
+  # If the instructor did not previously exist, create the section instructor relationship
+  # Returns true if the instructor could be added, false otherwise
+  public def add_instructor(user)
+    return false unless user.teacher?
+
+    Follower.find_by(section: self, student_user: user)&.destroy
+
+    si = SectionInstructor.with_deleted.find_by(instructor: user, section_id: id)
+    if si.blank?
+      SectionInstructor.create!(section_id: id, instructor: user, status: :active)
+    elsif si.deleted?
+      si.restore
+      si.status = :active
+      si.save!
+    elsif si.status != 'active'
+      si.status = :active
+      si.save!
+    end
```

Comment:
```suggestion
    transaction do
      Follower.find_by(section: self, student_user: user)&.destroy

      si = SectionInstructor.with_deleted.find_or_initialize_by(instructor: user, section_id: id)
      si.restore if si.deleted?
      si.active!
    end 
```

Reply:
The code change is just a suggestion, but please perform multiple mutation requests in a transaction.